app/services/staff_action_logger.rb

```diff
@@ -663,6 +663,29 @@ def log_embeddable_host(embeddable_host, action, opts = {})
     ))
   end
 
+  def log_api_key(api_key, action, opts = {})
+    opts[:changes]&.delete("key") # Do not log the full key
+
+    history_params = params(opts).merge(
+      action: action,
+      subject: api_key.truncated_key
+    )
+
+    if opts[:changes]&.keys == ["revoked_at", "updated_at"]
```

Comment:
This code block is quite confusing. Is there a more straightforward way to figure out if the key is being revoked/restored than inspecting the changes hash?

Reply:
You're right, this method got unnecessarily complicated. I've removed this logic and created separate functions for logging revoke/restore.